python/ray/workflow/tests/test_serialization.py

```diff
@@ -82,6 +84,26 @@ def test_dedupe_serialization_2(workflow_start_regular_shared):
     assert get_num_uploads() == 2
 
 
+def test_same_object_many_workflows(workflow_start_regular_shared):
+    """Ensure that when we dedupe uploads, we upload the object once per workflow,
```

Comment:
This means object ref can't be shared across jobs. I think it makes sense right now. Maybe we can think more about how to make it more efficient in the future.

Reply:
Yeah ideally we could share them across jobs, but we would have to worry a lot more about garbage collection.